manifests/params.pp

```diff
@@ -35,7 +35,13 @@
     }
   }
 
-  case $::lsbdistcodename {
+  if defined('$::lsbdistcodename') {
+    $_dist_code = $::lsbdistcodename
+  } else {
+    $_dist_code = 'default'
+  }
+
+  case $_dist_code {
```

Comment:
Maybe this whole case could be wrapped in an `if $::kernel == 'Linux'` or some such so we don't even bother with the case on non-linux boxes.  Otherwise, I think this is fine enough to fix the issue.  I'm seeing now other improvements we could make to this module.